automerge/src/types/opids.rs

```diff
@@ -0,0 +1,126 @@
+use itertools::Itertools;
+
+use super::OpId;
+
+/// A wrapper around `Vec<Opid>` which preserves the invariant that the ops are
+/// in ascending order with respect to their counters and actor IDs. In order to
+/// maintain this invariant you must provide a comparator function when adding
+/// ops as the actor indices in an  OpId are not sufficient to order the OpIds
+#[derive(Debug, Clone, PartialEq, Default)]
+pub(crate) struct OpIds(Vec<OpId>);
```

Comment:
From an API point of view, maybe it would be good to make `OpIds` have a generic for specifying the comparator func. Otherwise the same `OpIds` instance can be used with different comparators leading to bugs?

Reply:
That makes sense, thanks for explaining the rationale behind it.